### PR TITLE
[IMPROVEMENT] Local format in daySeparator

### PR DESCRIPTION
**Pull request by soundstorm**
Closing #1227

@RocketChat/ios

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/ChatViewController.swift [hunk 1]
```diff
@@ -855,7 +855,9 @@ extension ChatViewController {
             return UICollectionViewCell()
         }
 
-        cell.labelTitle.text = obj.timestamp.formatted("MMM dd, YYYY")
+        let dateFormatter = DateFormatter()
```
Comment (artrmz) [inline Rocket.Chat/Controllers/Chat/ChatViewController.swift]: Maybe let's create a singleton/static class that will hold formatter (one per format) and that will be responsible for formatting date? It's good practice to not unnecessarily create and destroy formatters that you're likely to need again especially in the cell's creation process.

Comment (soundstorm) [reply Rocket.Chat/Controllers/Chat/ChatViewController.swift]: Yeah, you're right

File: Rocket.Chat/Helpers/RCDateFormatter.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        dateFormatter.dateStyle = .none
+        dateFormatter.timeStyle = timeStyle
+        return self.dateFormatter.string(from: date)
+    }
+
+    static func datetime(_ date: Date, dateStyle: DateFormatter.Style = .medium, timeStyle: DateFormatter.Style = .short) -> String {
+        dateFormatter.dateStyle = dateStyle
+        dateFormatter.timeStyle = timeStyle
+        return self.dateFormatter.string(from: date)
+    }
```
Comment (rafaelks) [inline Rocket.Chat/Helpers/RCDateFormatter.swift]: Great job @soundstorm! What about having tests on this class? 🙈

Comment (rafaelks) [reply Rocket.Chat/Helpers/RCDateFormatter.swift]: @soundstorm I merged it already, to send another TestFlight, but can you please add tests in another PR?

**Pull request merged**